### ISSUE TITLE
Angel Nest is too cluttered with lava/water

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1226]
I've played the map Angels Nest today, and noticed what I've noticed on the other maps I played before: They are too cluttered with small patches of water/lava.

I like natural looking maps as much as the next guy, but the current maps overdo it to such an extent that I think they should be cleaned up a bit.
A minor issue is that it currently has water en lava tiles right next to each other, which does not look nice and really doesn't serve a purpose.

The major issue though is that it makes it really tedious for the player to tunnel somewhere, especially in combination with the fog of war and the huge map sizes. To see what I mean, load the map Angels Nest and try to tunnel all the way south from your starting dungeon, claiming the tunnel of course. (or load the replay from Issue #1224)
You'll notice that it requires constant player attention, because after every two tiles of digging you'll run into water/lava which you need to bridge. And only when you build the bridge can an imp claim two or 3 more tiles before you need to build another bridge for the imp to continue again. Repeat this around 40 (!) times before you reach the bottom.

A better approach to peppering the map with countless little puddles is to make fewer but larger lakes. And when you do make small patches of lava/water you should consider circling them with ground level dirt: This looks more natural and allows the player to simply dig the tunnel he wants because imps can automatically claim the tiles around it.

**Comment by hwoarangmy** [comment]
Looks straight forward. Note that this map is more a test of a tool that I did to convert an image into a map. There is no wanted gameplay here. If you see improvements, please feel free to edit the map and improve it.

**Comment by Bertram25** [comment]
@Loobinex  I precised the issue scope. Please don't open vague issue here, but rather open forums threads for such cases. Issues should have a defined scope, otherwise we will never be able to close them.

Is @hwoarangmy fix sufficient or do you see further improvements? Should the map be tagged as test?

**Comment by Loobinex** [comment]
@Betram25 The task at hand was not vague, just large and annoying, many of the included maps feature too much cluttered water/lava. Angel Nest was just a single example. The task would be to check and change all map.

But I can't be bothered by that either, so we'll leave it at this and if I play another map and I find it particularly annoying I'll report it for that specific map.

I've looked at the updated Angel Nest and the changes are sufficient for me yes, it's a lot better.

**Comment by Bertram25** [comment]
> The task at hand was not vague, just large and annoying

@Loobinex Please don't get me wrong. I fully understand the point here. the thing that I'm trying to defend as a developer standpoint, is that vague/large/neverending tasks never get done.

To keep our own motivation going and permit other to easily participate, we need to have itemized tasks so that someone can accomplish them in a sane amount of time. that's why I'm trying to advocate for large concept into forums and itemized tasks into the issues.

> Angel Nest was just a single example. The task would be to check and change all map.

I think you are right, while the tasks is even larger than being cluttered by lava/water, IMHO. As for the general stuff, I think it would be a good thing for people making levels to have a look at some level design articles, such as http://www.lesforges.org/article/level-design-1?scroll_to=article (Sorry, in french) (Disclaimer: I never said I was a good level designer at all, but it can't hurt.)

Level design is a neverending thing, as someone has to have knowledge in it, to get dedicated to that task, and have the right tool at hand before starting doing so. Definitely not a simple issue.

> But I can't be bothered by that either, so we'll leave it at this and if I play another map and I find it particularly annoying I'll report it for that specific map.
> I've looked at the updated Angel Nest and the changes are sufficient for me yes, it's a lot better.

Perfect, thanks! I'll close now.

**Comment by Loobinex** [comment]
@Bertram25 Having some high level placeholder issues also has some uses. If I would have taken my time, looked at all the maps with issues, and reported 25 separate issues for maps that need fixing your issue list will also start to look like a massive mountain to climb. I'm having no problem with this particular issue being closed, just talking in general.

'Big issues', even vague ones do have value on an issue list, and they don't have to hang around forever. They even help new people form a clear vision of what still needs to happen and how those things are prioritized. To have an issue like 'Make a campaign' or 'Get higher quality models' are perfectly fine and can hang around with the 'later' milestone until you decide that for the next release you actually want to work on that, at which point you can close it and open a range of smaller issues instead.

**Comment by hwoarangmy** [comment]
> Having some high level placeholder issues also has some uses

Yes but @Bertram25 is only saying that it looks more like forums issues (that should be on our forum) rather than github issues. I must say I agree with him here. That's because we use github issues to know more or less what everybody is working on and avoiding working on similar stuff when possible. Since there is not that much activity, it might seem a bit oversized but, a year ago, when there was more activity, it made some sense.

> To have an issue like 'Make a campaign' or 'Get higher quality models' are perfectly fine

For the campaign, yes, it may be an acceptable issue (that would be splitted into other issues like we do for the release issues).
For models, I don't think it is a good choice. Getting good quality models is obvious and usually, if we didn't, that's because we didn't found a better alternative. However, if the issue lists the models that should be improved, yes, it becomes acceptable to me.

Basically, the way we used to do, github issues were for contributors to pick up something they wanted to work on and to know what everybody is working on. To achieve that, issues must have a scope as small as possible.

**Comment by Bertram25** [comment]
@Loobinex I must say I'm fully aligned with @hwoarangmy there. As this permits to prevent cluttering the issue list per release, and get things done (TM).
That said, if general features and problems arise, forum can be there to display them, and issues can be there to split the job in sensitive steps.

> 'Make a campaign' or 'Get higher quality models' are perfectly fine and can hang around with the 'later' milestone until you decide that for the next release you actually want to work on that, at which point you can close it and open a range of smaller issues instead.

For instance, those are not fine, IMHO, as they will never ever be done as they are stated. Making a campaign and get higher quality models can always be on the go and never reached. Something certain people call 'Epics', btw.

**Comment by akien-mga** [comment]
I think it's fine to have broad issues at first that then get closed and repurposed into more specific issues. Whether it's on GitHub or on the forum is a matter of taste and policy, if we want this to stick on the forum it's fine by me. GH is IMO a bit more convenient for most developers discussions (in other projects I contribute to we even have a "discussion" label for such issues), as you can easily reference the discussion issue in more topical issues later on, but as mentioned it's a matter of taste.
